### PR TITLE
Add range with keyword arguments

### PR DESCRIPTION
**Pull request by ararslan**
Ref https://github.com/JuliaLang/julia/pull/25896

### CODE REVIEW

**Comment by ararslan** [comment]
Hm, the test failure is confusing.

File: src/Compat.jl [hunk 1]
```diff
@@ -1620,6 +1620,35 @@ if VERSION < v"0.7.0-DEV.3734"
...
+        if !(have_stop || have_length)
+            throw(ArgumentError("At least one of `length` or `stop` must be specified"))
+        elseif have_step && have_stop && have_length
+            throw(ArgumentError("Too many arguments specified; try passing only one of `stop` or `length`"))
+        elseif start === nothing
+            throw(ArgumentError("Can't start a range at `nothing`"))
+        end
+
+        if have_stop && !have_length
+            return have_step ? start:step:stop : start:stop
```
Comment (stevengj) [inline src/Compat.jl]: Need parens here.   Should be `have_step ? (start:step:stop) : (start:stop)` but actually parses as `have_step ? start : ((step:stop:start):stop)`

File: README.md [hunk 2]
```diff
@@ -278,6 +278,7 @@ Currently, the `@compat` macro supports the following syntaxes:
 
 * `Compat.IOBuffer` supporting keyword arguments ([#25873]).
 
+* `Compat.range` supporting keyword arguments ([25896]).
```
Comment (stevengj) [inline README.md]: Needs `#` in issue link.

File: src/Compat.jl (hunk 1 above)
Comment (ararslan) [reply src/Compat.jl]: image

**Pull request merged**